include/circt/Dialect/LTL/LTLTypes.td

```diff
@@ -0,0 +1,50 @@
+//===- LTLTypes.td - LTL dialect types ---------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_LTL_LTLTYPES_TD
+#define CIRCT_DIALECT_LTL_LTLTYPES_TD
+
+include "circt/Dialect/LTL/LTLDialect.td"
+include "mlir/IR/AttrTypeBase.td"
+
+class LTLTypeDef<string name, string typeMnemonic> : TypeDef<LTLDialect, name> {
+  let mnemonic = typeMnemonic;
+}
+
+def LTLSequenceType : LTLTypeDef<"Sequence", "sequence"> {
+  let summary = "LTL sequence type";
+  let description = [{
+    The `ltl.sequence` type is returned by any of the `ltl.seq.*` operations. It
+    represents a snippet of linear temporal logic, for example, *"A is true two
+    cycles after B is true"*.
+
+    Note that this type explicitly identifies a *sequence*. However, a boolean
+    value (`i1`) is also a valid sequence. Operations that accept a sequence as
+    an operand will use the `AnySequence` constraint, which also accepts `i1`.
+  }];
+}
+
+def LTLPropertyType : LTLTypeDef<"Property", "property"> {
```

Comment:
Are we defining properties? Or propositions? I think 'proposition' sounds slightly more precise.

Reply:
I'm trying to stick with the SVA spec for starters until we have built up a bit of know-how and experience with the dialect. I don't fully understand what "proposition" entails. It seems to be used in LTL specifically to refer to the atomic boolean propositions that you compose with the LTL operators, which is a different thing than the properties here (which are the result of applying the operators).